File: Python/Endpoint Examples/JSON Payload/markdown.py
import requests
import json

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(
    url='https://api.pdfrest.com/upload',
    data=upload_data,
    headers={
        'Content-Type': 'application/octet-stream',
        'Content-Filename': 'file.pdf',
        'API-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
    }
)

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent=2))

    uploaded_id = upload_response_json['files'][0]['id']
    markdown_data = { "id": uploaded_id, "page_break_comments": "on" }
    print(json.dumps(markdown_data, indent=2))

    print("Processing file...")
    markdown_response = requests.post(
        url='https://api.pdfrest.com/markdown',
        data=json.dumps(markdown_data),
        headers={
            'Content-Type': 'application/json',
            'API-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
        }
    )

    print("Processing response status code: " + str(markdown_response.status_code))
    if markdown_response.ok:
        markdown_response_json = markdown_response.json()
        print(json.dumps(markdown_response_json, indent=2))
    else:
        print(markdown_response.text)
else:
    print(upload_response.text)

File: JavaScript/Endpoint Examples/JSON Payload/markdown.js
var axios = require("axios");
var fs = require("fs");

// By default, we use the US-based API service. This is the primary endpoint for global use.
var apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//var apiUrl = "https://eu-api.pdfrest.com";

var upload_data = fs.createReadStream("/path/to/file.pdf");

var upload_config = {
  method: "post",
  maxBodyLength: Infinity,
  url: apiUrl + "/upload",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
    "Content-Filename": "filename.pdf",
    "Content-Type": "application/octet-stream",
  },
  data: upload_data,
};

// Send upload request
axios(upload_config)
  .then(function (upload_response) {
    console.log("Upload response:");
    console.log(JSON.stringify(upload_response.data, null, 2));

    var uploaded_id = upload_response.data.files[0].id;

    var markdown_config = {
      method: "post",
      maxBodyLength: Infinity,
      url: apiUrl + "/markdown",
      headers: {
        "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
        "Content-Type": "application/json",
      },
      data: {
        id: uploaded_id,
        page_break_comments: "on"
      },
    };

    // Send markdown request
    axios(markdown_config)
      .then(function (markdown_response) {
        console.log("Markdown response:");
        console.log(JSON.stringify(markdown_response.data, null, 2));
      })
      .catch(function (error) {
        console.error("Markdown request error:");
        console.error(error.response?.data || error.message);
      });
  })
  .catch(function (error) {
    console.error("Upload request error:");
    console.error(error.response?.data || error.message);
  });
